cpp/lib/ome/common/units/length.h

```diff
@@ -0,0 +1,604 @@
+      BOOST_UNITS_STATIC_CONSTANT(parsecs, parsec_unit);
+      /// Measured quantity in parsecs.
+      typedef quantity<parsec_unit> parsec_quantity;
+
+      /// Unit definition for point length.
+      typedef scaled_base_unit<boost::units::imperial::inch_base_unit, scale<72, static_rational<-1> > >::unit_type point_unit;
+      /// Numeric constant for point.
+      BOOST_UNITS_STATIC_CONSTANT(point, point_unit);
+      /// Numeric constant for point.
+      BOOST_UNITS_STATIC_CONSTANT(points, point_unit);
+      /// Measured quantity in points.
+      typedef quantity<point_unit> point_quantity;
+
+      /// Base unit for pixel length.
+      struct pixel_base_unit : boost::units::base_unit<pixel_base_unit,
+                                                       boost::units::length_dimension, 1> { };
+      /// Unit system for pixel length.
+      typedef boost::units::make_system<
+        pixel_base_unit>::type pixel_system;
+      /// Unit definition for pixel length (undefined length, not convertible to other length units).
+      typedef boost::units::unit<boost::units::length_dimension, pixel_system> pixel_unit;
+      /// Numeric constant for pixel.
+      BOOST_UNITS_STATIC_CONSTANT(pixel, pixel_unit);
+      /// Numeric constant for pixel.
+      BOOST_UNITS_STATIC_CONSTANT(pixels, pixel_unit);
+      /// Measured quantity in pixels.
+      typedef quantity<pixel_unit> pixel_quantity;
+
+      /// Base unit for reference frame.
+      struct reference_frame_base_unit : boost::units::base_unit<reference_frame_base_unit,
+                                                                 boost::units::length_dimension, 2> { };
+      /// Unit system for reference frame length.
+      typedef boost::units::make_system<
+        reference_frame_base_unit>::type reference_frame_system;
+      /// Unit definition for reference frame unit length (undefined unit length, not convertible to other length units).
+      typedef boost::units::unit<boost::units::length_dimension, reference_frame_system> reference_frame_unit;
+      /// Numeric constant for reference frame unit.
+      BOOST_UNITS_STATIC_CONSTANT(reference_frame, reference_frame_unit);
+      /// Measured quantity in reference frame units.
+      typedef quantity<reference_frame_unit> reference_frame_quantity;
+
+    }
+  }
+}
+
+namespace boost
+{
+  namespace units
+  {
+
+    /// Unit information for pixels.
+    template<> struct base_unit_info<ome::common::units::pixel_base_unit>
+    {
+      /// Unit name.
+      static std::string name()   { return "pixel"; }
+      /// Unit symbol.
+      static std::string symbol() { return "px"; }
+    };
+
+    /// Unit information for reference frame.
+    template<> struct base_unit_info<ome::common::units::reference_frame_base_unit>
+    {
+      /// Unit name.
+      static std::string name()   { return "reference frame"; }
+      /// Unit symbol.
+      static std::string symbol() { return "r.f."; }
```

Comment:
Stream output.

```
reference_frame_quantity q(4.0 * reference_frame);
std::cout << q << '\n';
```

Note that these are *not* the unit enumerations defined in the OME data model; they are abbreviations required by the underlying units library.  The actual model units support will match the model exactly.  This unit symbol can be changed if needed, though it's not indended to be user-visible except perhaps in debug logging.